grails-app/services/org/pih/warehouse/requisition/RequisitionService.groovy

```diff
@@ -825,23 +825,34 @@ class RequisitionService {
         return new Event(eventDate: eventDate, eventType: eventType, eventLocation: eventLocation, createdBy: currentUser)
     }
 
+    Requisition transitionRequisitionStatus(Requisition requisition, RequisitionStatus requisitionStatus, EventCode eventCode, User currentUser) {
+        requisition.status = requisitionStatus
+        Event event = createEvent(eventCode, requisition.origin, new Date(), currentUser)
+        requisition.addToEvents(event)
+        requisition.save()
+    }
+
     void triggerRequisitionStatusTransition(Requisition requisitionInstance, User currentUser) {
         Requisition.withSession {
             Requisition requisition = Requisition.get(requisitionInstance.id)
             if (requisition.status == RequisitionStatus.PENDING_APPROVAL) {
```

Comment:
What happens if we call triggerRequisitionStatusTransition() multiple times with PENDING_APPROVAL, APPROVED, REJECTED? Is there any protection against that?